addons/binding/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/motionsensor/DLinkMotionSensorCommunication.java

```diff
@@ -0,0 +1,434 @@
+        if (next.isAfter(nextStart)) {
+            next = nextStart;
+        }
+
+        return now.until(next, ChronoUnit.MILLIS);
+    }
+
+    /**
+     * Output unexpected responses to the debug log and sets the FIRMWARE error.
+     *
+     * @param message
+     * @param soapResponse
+     */
+    private void unexpectedResult(final String message, final Document soapResponse) {
+        logUnexpectedResult(message, soapResponse);
+
+        // Best guess when receiving unexpected responses
+        status = DeviceStatus.UNSUPPORTED_FIRMWARE;
+    }
+
+    /**
+     * Sends the two login messages and sets the authentication header for the action
+     * message.
+     *
+     * @param action
+     * @param timeout
+     */
+    private void login(final SOAPMessage action, final int timeout) {
+        loginSuccess = false;
+
+        login(timeout);
+        setAuthenticationHeaders(action);
+
+        switch (getHNAPStatus()) {
+            case LOGGED_IN:
+                loginSuccess = true;
+                break;
+            case COMMUNICATION_ERROR:
+                status = DeviceStatus.COMMUNICATION_ERROR;
+                break;
+            case INVALID_PIN:
+                status = DeviceStatus.INVALID_PIN;
+                break;
+            case INTERNAL_ERROR:
+                status = DeviceStatus.INTERNAL_ERROR;
+                break;
+            case UNSUPPORTED_FIRMWARE:
+                status = DeviceStatus.UNSUPPORTED_FIRMWARE;
+                break;
+            case INITIALISED:
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Sends the detection message
+     *
+     * @param isRetry - Has this been called as a result of a login retry
+     * @return true, if the last detection time was successfully retrieved, otherwise false
+     */
+    private boolean getLastDetection(final boolean isRetry) {
+        detectSuccess = false;
+
+        if (loginSuccess) {
+            try {
+                final Document soapResponse = sendReceive(detectionAction, DETECT_TIMEOUT_MS);
+
+                final Node result = soapResponse.getElementsByTagName("GetLatestDetectionResult").item(0);
+
+                if (result != null) {
+                    if (OK.equals(result.getTextContent())) {
+                        final Node timeNode = soapResponse.getElementsByTagName("LatestDetectTime").item(0);
+
+                        if (timeNode != null) {
+                            prevDetection = lastDetection;
+                            lastDetection = Long.valueOf(timeNode.getTextContent());
+                            detectSuccess = true;
+                        } else {
+                            unexpectedResult("getLastDetection - Unexpected response", soapResponse);
+                        }
+                    } else if (isRetry) {
+                        unexpectedResult("getLastDetection - Unexpected response", soapResponse);
+                    }
```

Comment:
Yes that's expected as I only want to log the unexpectedResult when its a retry (a retry is attempted in case the login session has timed out),